### PR TITLE
module and edge samples

### PR DESCRIPTION
**Pull request by BertKleewein**
Sample code for Modules, including a dockerfile and some docker build scripts which will soon be out of date (as soon as I can release some -preview NPM packages)

### CODE REVIEW

File: device/edge-sample/edge_sample_module.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+var client = Client.fromConnectionString(connectionString, Protocol);
+console.log('got client');
+
+client.on('error', function (err) {
+  console.error(err.message);
+});
+
+client.setOptions({
+  ca: fs.readFileSync(process.env.EdgeModuleCACertificateFile).toString('ascii')
```
Comment (pierreca) [inline device/edge-sample/edge_sample_module.js]: >EdgeModuleCACertificateFile [](start = 34, length = 27)

nit: that's unusual casing/format for an env variable. usually see them all uppercase with underscores EDGE_CA_CERT for example

File: device/edge-sample/readme.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Microsoft Azure IoT SDK for Node.js Edge module sample
+
+This folder contains a set of Dockerfiles which can be used to crate an Azure IoT Edge module using node.js which can be deployed and run inside of an Azure IoT Edge environment
```
Comment (pierreca) [inline device/edge-sample/readme.md]: >crate [](start = 63, length = 5)

typo (create)

File: device/edge-sample/readme.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+## Implementation details
+
+Because our build system relies heavily on the npm link command, and because our current code isn't available in the npm registry just yet, it is necessary to jump through a few (hopefully small) hoops to make this container work.
+
+We do this by first building a base image which contains the azure-iot-sdk-node code from your current clone of the repository.  We do this once, assuming this code won't change.  If you do make changes to the various azure modules, or if you pull a new set of changes down, you will need to re-build the base image.  Otherwise, if the azure-iot-sdk-node code doesn't change, you can build it and forget about it.
+
+The base imagew is perfectly usable as an edge module.  You can tag this and push it to your Docker image registry and it should work as is.
```
Comment (pierreca) [inline device/edge-sample/readme.md]: >imagew [](start = 9, length = 6)

typo (w at the end of image)

File: readme-modules.md [hunk 4]
```diff
@@ -7,18 +7,11 @@ This code is not currently published on NPM.  In order to use these modules, you
...
-
-* 'hubName' is the name of your IoT hub, possibly in the azure-devices.net domain. (e.g. foo.azure-devices.net)
-* 'deviceId' is the name of your IoT Edge Device
-* 'moduleId' is the name of your Edge Module
-* 'key' is the shared access key for your module
-* 'edgeIp' is the IP address or DNS name for the machine hosting the edgeHub instance for the 'deviceId' device.  TCP port 8883 must be open on this device and be routed to the container hosting the edgeHub module.
-
-Currently, a connection string can be obtained using the Bootstrapping instructions below.
+### Recent updates:
+* It is no longer to extract the connection string from an existing module.
```
Comment (pierreca) [inline readme-modules.md]: >no longer [](start = 8, length = 9)

no longer... Needed? missing word?

File: device/edge-sample/edge_sample_module.js (hunk 1 above)
Comment (BertKleewein) [reply device/edge-sample/edge_sample_module.js]: That's the convention that they use in Edge containers.  I don't like it either, but I didn't invent it.

---
In reply to: [178390273](https://github.com/Azure/azure-iot-sdk-node/pull/260#discussion_r178390273) [](ancestors = 178390273)

**Pull request closed without merging**